omod/src/main/java/org/openmrs/module/adminui/page/controller/account/ChangeDefaultsPageController.java

```diff
@@ -0,0 +1,146 @@
+			User user = Context.getAuthenticatedUser();
+			Map<String, String> props = user.getUserProperties();
+			UserDefaults userDefaults = new UserDefaults();
+			userDefaults.setDefaultLocale(props.get(OpenmrsConstants.USER_PROPERTY_DEFAULT_LOCALE));
+			userDefaults.setProficientLocales(props.get(OpenmrsConstants.USER_PROPERTY_PROFICIENT_LOCALES));
+			pageModel.addAttribute("userDefaults", userDefaults);
+			pageModel.addAttribute("locales", Context.getAdministrationService().getPresentationLocales());
+		}
+		return "account/changeDefaults";
+	}
+
+	public String post(@MethodParam("createUserDefaults") @BindParams UserDefaults defaults,
+	                   BindingResult errors,
+	                   @SpringBean("userService") UserService userService,
+	                   @SpringBean("messageSourceService") MessageSourceService messageSourceService,
+	                   @SpringBean("messageSource") MessageSource messageSource,
+	                   HttpServletRequest request,
+	                   PageModel model) {
+		if (errors.hasErrors()) {
+			sendErrorMessage(errors, messageSource, request);
+			model.addAttribute("errors", errors);
+			return "account/changePassword";
+		}
+		return saveDefaults(defaults, userService, messageSourceService, request);
+	}
+
+	private void sendErrorMessage(BindingResult errors, MessageSource messageSource, HttpServletRequest request) {
+		List<ObjectError> allErrors = errors.getAllErrors();
+		String message = getMessageErrors(messageSource, allErrors);
+		request.getSession().setAttribute(AdminUiConstants.SESSION_ATTRIBUTE_ERROR_MESSAGE, message);
+	}
+
+	private String getMessageErrors(MessageSource messageSource, List<ObjectError> allErrors) {
+		String message = "";
+		for (ObjectError error : allErrors) {
+			Object[] arguments = error.getArguments();
+			String errorMessage = messageSource.getMessage(error.getCode(), arguments, Context.getLocale());
+			message = message.concat(replaceArguments(errorMessage, arguments).concat("<br>"));
+		}
+		return message;
+	}
+
+	private String replaceArguments(String message, Object[] arguments) {
+		if (arguments != null) {
+			for (int i = 0; i < arguments.length; i++) {
+				String argument = (String) arguments[i];
+				message = message.replaceAll("\\{" + i + "\\}", argument);
+			}
+		}
+		return message;
+	}
+
+	public String saveDefaults(UserDefaults defaults, UserService userService, MessageSourceService messageSourceService,
+	                           HttpServletRequest request) {
+		try {
+			User user = Context.getAuthenticatedUser();
+			Map<String, String> props = user.getUserProperties();
+			props.put(OpenmrsConstants.USER_PROPERTY_DEFAULT_LOCALE, defaults.getDefaultLocale());
+			props.put(OpenmrsConstants.USER_PROPERTY_PROFICIENT_LOCALES, defaults.getProficientLocales());
+			user.setUserProperties(props);
+			userService.saveUser(user, null);
+			request.getSession().setAttribute(AdminUiConstants.SESSION_ATTRIBUTE_INFO_MESSAGE,
```

Comment:
Sorry, I totally don't understand what these lines have to do with InfoErrorMessageUtil.flashInfoMessage(..). :)